internal/security/grantvmgroupaccess_test.go

```diff
@@ -0,0 +1,291 @@
+		},
+		{
+			name:            "Write",
+			access:          AccessMaskWrite,
+			filePermissions: []string{`(I)(W,Rc)`},
+			dirPermissions:  []string{`(W,Rc)`, `(OI)(CI)(IO)(GW)`},
+		},
+		{
+			name:            "Execute",
+			access:          AccessMaskExecute,
+			filePermissions: []string{`(I)(Rc,S,X,RA)`},
+			dirPermissions:  []string{`(Rc,S,X,RA)`, `(OI)(CI)(IO)(GE)`},
+		},
+		{
+			name:            "ReadWrite",
+			access:          AccessMaskRead | AccessMaskWrite,
+			filePermissions: []string{`(I)(R,W)`},
+			dirPermissions:  []string{`(R,W)`, `(OI)(CI)(IO)(GR,GW)`},
+		},
+		{
+			name:            "ReadExecute",
+			access:          AccessMaskRead | AccessMaskExecute,
+			filePermissions: []string{`(I)(RX)`},
+			dirPermissions:  []string{`(RX)`, `(OI)(CI)(IO)(GR,GE)`},
+		},
+		{
+			name:            "WriteExecute",
+			access:          AccessMaskWrite | AccessMaskExecute,
+			filePermissions: []string{`(I)(W,Rc,X,RA)`},
+			dirPermissions:  []string{`(W,Rc,X,RA)`, `(OI)(CI)(IO)(GW,GE)`},
+		},
+		{
+			name:            "ReadWriteExecute",
+			access:          AccessMaskRead | AccessMaskWrite | AccessMaskExecute,
+			filePermissions: []string{`(I)(RX,W)`},
+			dirPermissions:  []string{`(RX,W)`, `(OI)(CI)(IO)(GR,GW,GE)`},
+		},
+		{
+			name:            "All",
+			access:          AccessMaskAll,
+			filePermissions: []string{`(I)(F)`},
+			dirPermissions:  []string{`(F)`, `(OI)(CI)(IO)(F)`},
+		}} {
+		t.Run(cfg.name, func(t *testing.T) {
+			dir := t.TempDir()
+			fd, err := os.Create(filepath.Join(dir, "test.txt"))
+			if err != nil {
+				t.Fatalf("failed to create temporary file: %s", err)
+			}
+			defer func() {
+				_ = fd.Close()
+				_ = os.Remove(fd.Name())
+			}()
+
+			if err := GrantVmGroupAccessWithMask(dir, cfg.access); err != nil {
+				t.Fatal(err)
+			}
+			verifyVMAccountDACLs(t, dir, cfg.dirPermissions)
+			verifyVMAccountDACLs(t, fd.Name(), cfg.filePermissions)
+		})
+	}
+}
+
+func TestGrantVmGroupAccess_Invalid_AccessMask(t *testing.T) {
+	for _, access := range []AccessMask{
+		0,          // no bits set
+		1,          // invalid bit set
+		0x02000001, // invalid extra bit set
+	} {
+		t.Run(fmt.Sprintf("AccessMask_0x%x", access), func(t *testing.T) {
+			dir := t.TempDir()
+			fd, err := os.Create(filepath.Join(dir, "test.txt"))
+			if err != nil {
+				t.Fatalf("failed to create temporary file: %s", err)
+			}
+			defer func() {
+				_ = fd.Close()
+				_ = os.Remove(fd.Name())
+			}()
+
+			if err := GrantVmGroupAccessWithMask(fd.Name(), access); err == nil {
+				t.Fatalf("expected an error for mask: %x", access)
+			}
+		})
+	}
+}
+
+func verifyVMAccountDACLs(t *testing.T, name string, permissions []string) {
+	cmd := exec.Command("icacls", name)
+	outb, err := cmd.CombinedOutput()
+	if err != nil {
+		t.Fatal(err)
+	}
+	out := string(outb)
+
+	for _, p := range permissions {
+		// Avoid '(' and ')' being part of match groups
+		p = strings.Replace(p, "(", "\\(", -1)
```

Comment:
`strings.ReplaceAll(`
Also, `https://pkg.go.dev/regexp#QuoteMeta`, may do what you want.

Reply:
used `QuoteMeta`